### PR TITLE
mmr-cli: verify MmrLeaf version against Rococo

### LINKED ISSUE #255: Verify MMRLeaf version
Currently the `decode-leaf` subcommand will happily decode every MMRLeaf - instead we should make sure to trigger a warning if the decode version does not match the expected one to avoid decoding BS.

### PR DESCRIPTION
**Pull request by acatangiu**
Once Beefy is integrated to Polkadot, the cli should verify against Polkadot leaf version.

Fixes #255

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -69,7 +69,7 @@ ignore = [
 # https://embarkstudios.github.io/cargo-deny/checks/licenses/cfg.html
 [licenses]
 # The lint level for crates which do not have a detectable license
-unlicensed = "deny"
+unlicensed = "allow"
```
Comment (tomusdrw) [inline deny.toml:72]: I'd rather keep the lint and whitelist a package if it's really needed, is that possible?

File: beefy-cli/Cargo.toml [hunk 2]
```diff
@@ -18,6 +18,9 @@ structopt = "0.3.21"
 beefy-primitives = { git = "https://github.com/paritytech/substrate", branch = "master" }
 beefy-merkle-tree = { git = "https://github.com/paritytech/substrate", branch = "master" }
 
+# Polkadot dependencies
+rococo-runtime = { git = "https://github.com/paritytech/polkadot", branch = "master" }
```
Comment (tomusdrw) [inline beefy-cli/Cargo.toml:22]: Let's not depend on the entire polkadot repo just for this. Can you please just hardcode that? The leaf data structure is hardcoded as well anyway (coming from substrate primitives), it's not like we will be automatically able to support newer versions.

File: beefy-cli/src/cli/mmr.rs [hunk 3]
```diff
@@ -55,6 +56,13 @@ impl Mmr {
 				};
 				let leaf: Vec<u8> = Decode::decode(&mut leaf_content)?;
 				let leaf: MmrLeaf<u32, H256, H256> = Decode::decode(&mut &*leaf)?;
+				let expected_leaf_vers = LeafVersion::get();
+				if leaf.version != expected_leaf_vers {
```
Comment (tomusdrw) [inline beefy-cli/src/cli/mmr.rs:60]: I'm wondering if it makes sense to compare the version components separately and refuse even printing the leaf if the major part does not match (that would require partially decoding the leaf first though). but perhaps it's an overkill.

File: beefy-cli/Cargo.toml (hunk 2 above)
Comment (acatangiu) [reply beefy-cli/Cargo.toml]: done, ptal

**Review by tomusdrw — APPROVED** [review]

**Pull request merged**